telemetry/transformers/payload.go

```diff
@@ -0,0 +1,79 @@
+package transformers
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/teslamotors/fleet-telemetry/protos"
+)
+
+func PayloadToMap(payload *protos.Payload, excludeTypes bool) (map[string]interface{}, error) {
+	convertedPayload := make(map[string]interface{}, len(payload.Data)+2)
+	convertedPayload["Vin"] = payload.Vin
+	convertedPayload["CreatedAt"] = payload.CreatedAt.AsTime().Format(time.RFC3339)
+
+	var err error
+	for _, datum := range payload.Data {
+		name := protos.Field_name[int32(datum.Key.Number())]
+		convertedPayload[name], err = transformValue(datum.Value.Value, excludeTypes)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return convertedPayload, nil
+}
+
+func transformValue(value interface{}, excludeTypes bool) (interface{}, error) {
+	var outputValue interface{}
+	var outputType string
+
+	// ordered by expected frequency
+	switch v := value.(type) {
+	case *protos.Value_StringValue:
+		outputType = "stringValue"
+		outputValue = v.StringValue
+	case *protos.Value_LocationValue:
+		outputType = "locationValue"
+		outputValue = map[string]float64{
+			"latitude":  v.LocationValue.Latitude,
+			"longitude": v.LocationValue.Longitude,
+		}
+	case *protos.Value_FloatValue:
+		outputType = "floatValue"
+		outputValue = v.FloatValue
+	case *protos.Value_IntValue:
+		outputType = "intValue"
+		outputValue = v.IntValue
+	case *protos.Value_DoubleValue:
+		outputType = "doubleValue"
+		outputValue = v.DoubleValue
+	case *protos.Value_LongValue:
+		outputType = "longValue"
+		outputValue = v.LongValue
+	case *protos.Value_BooleanValue:
+		outputType = "booleanValue"
+		outputValue = v.BooleanValue
+	case *protos.Value_Invalid:
+		outputType = "invalid"
+		if excludeTypes {
+			outputValue = "<invalid>"
+		} else {
+			outputValue = true
+		}
+	case *protos.Value_ShiftStateValue:
+		outputType = "shiftStateValue"
+		outputValue = v.ShiftStateValue.String()
+	case *protos.Value_ChargingValue:
+		outputType = "chargingValue"
+		outputValue = v.ChargingValue.String()
+	default:
+		return nil, fmt.Errorf("unknown value type in payload: %T", value)
```

Comment:
instead of returning error, should we try to just return output value as string?